sklearn/multiclass.py

```diff
@@ -540,16 +563,14 @@ def predict(self, X):
                 votes[pred == 1, j] += 1
                 k += 1
 
-        # find all places with maximum votes per sample
-        maxima = votes == np.max(votes, axis=1)[:, np.newaxis]
+        max_scores = scores.max()
+        min_scores = scores.min()
 
-        # if there are ties, use scores to break them
-        if np.any(maxima.sum(axis=1) > 1):
-            scores[~maxima] = -np.inf
-            prediction = scores.argmax(axis=1)
-        else:
-            prediction = votes.argmax(axis=1)
-        return self.classes_[prediction]
+        if max_scores == min_scores:
+            return votes
+
+        # Scale the scores to [-0.4, 0.4] and add it with votes
+        return votes + scores * (0.4 / max(abs(max_scores), abs(min_scores)))
```

Comment:
adding an eps to the divisor is required to avoid a division by zero in the pathological case where all the base estimators return a score of 0 for all decisions.